### DISCUSSION TITLE
关于docker的使用求助

### DISCUSSION THREAD

**Discussion by boli-design** [docker-easyconnect/docker-easyconnect#39] [Q&A] [answered]
首先感谢大佬的辛勤付出。我使用的是win10系统，看教程基本没法跟上步骤。大佬的教程是基于linux的吗？还是基于windows？使用docker的部分有没有具体一点的操作步骤或是教程呢？

**Comment by Hagb** [comment]
Hi，请问看的是什么教程，什么步骤上遇到了问题？（但我不了解 Windows 上的操作）

这个仓库里的说明文档编写时确实是基于 GNU/Linux 的，但是镜像本身在 Windows 的 docker 上应该也可以使用。

或许可以参考官方文档 https://docs.docker.com/docker-for-windows/install/ 或在搜索引擎上搜索“docker windows”

**Comment by boli-design** [comment]
感谢回复。官方的教程估计需要啃一段时间才有起色。
最直观的问题是：第二步，在终端输入balabala，终端指的是docker中的输入窗口吗，这个仅在刚安装好后出现过，就再也找不到了；第三步：使用vnc客户端连接vnc，这里的vnc指的是docker中的组件吗，还是其他的连接工具？；第三步：地址：127.0.0.1, 端口: 5901, 密码 xxxx，密码就填“xxxx”吗？
目前仅做到第二步终端输入（且不确定是否正确），还没有找到第三步的入口

**Comment by Hagb** [comment]
第二步的输出是什么？
第三步是需要使用 vnc 客户端来连接图形界面以操作 easyconnect。

**Comment by boli-design** [comment]
第二步之后，我获得了一个界面：
![image](https://user-images.githubusercontent.com/57988442/107876906-e6a39f80-6f03-11eb-92f9-720fbcb0119b.png)

![image](https://user-images.githubusercontent.com/57988442/107876909-f1f6cb00-6f03-11eb-84ca-2812a211faff.png)
处于running状态，但不知如何处理

**Comment by Hagb** [comment]
这看起来没问题啊。（虽然我没用过 windows 版）
之后你需要用一个 vnc 客户端来操作 easyconnect

**Comment by boli-design** [comment]
目前下载了某个vnc客户端，地址输入127.0.0.1，但显示连接被拒绝
![image](https://user-images.githubusercontent.com/57988442/107909225-bc4cf300-6f92-11eb-8be0-b1523e3098f5.png)

查找了正在运行的hagb/docker-easyconnect:latest 日志，如下所示：
![image](https://user-images.githubusercontent.com/57988442/107909461-33828700-6f93-11eb-8e50-9788fa0419b4.png)

**Comment by Hagb** [comment]
> 目前下载了某个vnc客户端，地址输入127.0.0.1，但显示连接被拒绝 

那加上端口号呢？ 

> 查找了正在运行的hagb/docker-easyconnect:latest 日志，如下所示：
> ![image](https://user-images.githubusercontent.com/57988442/107909461-33828700-6f93-11eb-8e50-9788fa0419b4.png)

这报错看起来不太正常…… 你用来启动的命令具体是什么？

**Comment by boli-design** [comment]
加上端口号后，依旧显示无法连接
![image](https://user-images.githubusercontent.com/57988442/108164352-94dd5e00-712b-11eb-95d4-724bf318d43a.png)
启动时，在CMD窗口中输入命令：
docker run --device /dev/net/tun --cap-add NET_ADMIN -ti -e PASSWORD=xxxx -v $HOME/.ecdata:/root -p 127.0.0.1:5901:5901 -p 127.0.0.1:1080:1080 hagb/docker-easyconnect
之后便自动生成容器，并运行了

**Comment by Hagb** [comment, marked as answer]
solved in #49
